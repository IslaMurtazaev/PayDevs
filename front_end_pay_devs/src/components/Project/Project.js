import { connect } from "react-redux";
import { projectActions } from "../../actions/project";

import ProjectScreen from "./ProjectScreen";

const mapStateToProps = (state, ownProps) => ({
  project: state.project
    ? state.project
    : state.projects.find(project => project.id === +ownProps.match.params.id)
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  removeProject: id => {
    dispatch(projectActions.remove(id));
  },
  getTotal: id => {
    dispatch(projectActions.getTotal(id));
  },
  getProject: () => {
    dispatch(projectActions.get(+ownProps.match.params.id));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ProjectScreen);
