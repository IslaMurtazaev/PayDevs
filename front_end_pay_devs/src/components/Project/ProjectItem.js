import React, { Component } from "react";
import { connect } from "react-redux";
import { projectActions } from "../../actions/project";
import { Link } from "react-router-dom";
import Tasks from "../../components/Task/Tasks";
import HourPayments from "../../components/HourPayment/HourlyRates";
import MonthRates from "../MonthPayment/MonthlyRates";

class ProjectItem extends Component {
  componentDidMount() {
    if (!this.props.project.title) this.props.getProject();
  }

  render() {
    let { project, getTotal, removeProject } = this.props;

    let sessionsType;
    let type_of_payment;
    switch (project.type_of_payment) {
      case "M_P":
        type_of_payment = "Monthly";
        sessionsType = <MonthRates projectId={project.id} />;
        break;
      case "H_P":
        type_of_payment = "Hourly";
        sessionsType = <HourPayments projectId={project.id} />;
        break;
      case "T_P":
        type_of_payment = "Taskly";
        sessionsType = <Tasks projectId={project.id} />;
        break;
      default:
        break;
    }

    return (
      <div className="container">
        <h1 className="projectTitle">{project.title}</h1>
        <div className="properties">
          {project.description ? (
            <h4 className="projectDescription">
              <span className="property">Description: </span>
              {project.description}
            </h4>
          ) : null}
          {project.start_date ? (
            <h4 className="projectStartDate">
              <span className="property">Start date: </span>
              {new Date(project.start_date).toDateString()}
            </h4>
          ) : null}
          {project.end_date ? (
            <h4 className="projectEndDate">
              <span className="property">End date: </span>
              {new Date(project.end_date).toDateString()}
            </h4>
          ) : null}
          <h4 className="projectTypeOfPayment">
            <span className="property">Type of payment:</span> {type_of_payment}
          </h4>
          <h4 className="projectStatus">
            <span className="property">Status:</span>
            {project.status ? "" : "not"} active
          </h4>
        </div>

        <div className="btn-group-horizontal">
          <button
            type="button"
            className="btn btn-success btn-lg"
            onClick={() => getTotal(project.id)}
          >
            Total
          </button>

          <Link to={`${project.id}/update`}>
            <button type="button" className="btn btn-warning btn-lg">
              Update project
            </button>
          </Link>

          <button
            type="button"
            className="removeProject btn btn-danger btn-lg"
            onClick={() => removeProject(project.id)}
          >
            Delete project
          </button>
        </div>

        <div className="session">{sessionsType}</div>
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => ({
  project: state.project
    ? state.project
    : state.projects.find(project => project.id === +ownProps.match.params.id)
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  removeProject: id => {
    dispatch(projectActions.remove(id));
  },
  getTotal: id => {
    dispatch(projectActions.getTotal(id));
  },
  getProject: () => {
    dispatch(projectActions.get(+ownProps.match.params.id));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ProjectItem);
