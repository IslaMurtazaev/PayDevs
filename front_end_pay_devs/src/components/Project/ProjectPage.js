import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { projectActions } from "../../actions/project";

class ProjectPage extends Component {
  componentWillMount() {
    if (!this.props.projects.length) this.props.getAllProjects();
  }

  render() {
    const { user } = this.props.user; // why we are wrapping user inside a user?
    const { projects } = this.props;
    return (
      <div>
        <div className="user">
          {user && <div className="username">{user.username}</div>}
          {user && <div className="email">{user.email}</div>}
        </div>
        <Link className="newProjectLink" to="/project/create">
          New Project
        </Link>
        <div className="projectList">
          <ul>
            {projects.map(project => (
              <li key={project.id}>
                <div className="projectInfo">
                  <Link className="projectLink" to={`/project/${project.id}`}>
                    {project.title}
                  </Link>
                  <p>{new Date(project.start_date).toDateString()}</p>
                </div>
              </li>
            ))}
          </ul>
        </div>
      </div>
    );
  }
}

export default connect(
  state => ({
    user: state.user,
    projects: state.projects
  }),
  dispatch => ({
    getAllProjects: () => {
      dispatch(projectActions.getAll());
    }
  })
)(ProjectPage);
