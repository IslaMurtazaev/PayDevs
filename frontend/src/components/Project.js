import React, { Component } from "react";
import { Link } from "react-router-dom";

import Tasks from "../containers/Tasks";
import HourlyRates from "../containers/HourlyRates";
import MonthlyRates from "../containers/MonthlyRates";

class Project extends Component {
  componentWillMount() {
    if (this.props.project.id !== +this.props.match.params.id)
      this.props.getProject();
  }

  render() {
    let { project, getTotal, removeProject } = this.props;
    let projectId = +this.props.match.params.id;

    let sessionsType;
    let type_of_payment;
    switch (project.type_of_payment) {
      case "M_P":
        type_of_payment = "Monthly";
        sessionsType = <MonthlyRates projectId={projectId} />;
        break;
      case "H_P":
        type_of_payment = "Hourly";
        sessionsType = <HourlyRates projectId={projectId} />;
        break;
      case "T_P":
        type_of_payment = "Taskly";
        sessionsType = <Tasks projectId={projectId} />;
        break;
      default:
        break;
    }

    return (
      <div className="container">
        <h1 className="projectTitle">{project.title}</h1>
        <div className="properties">
          {project.description ? (
            <p className="projectDescription">
              <span className="property">Description: </span>
              {project.description}
            </p>
          ) : null}
          {project.start_date ? (
            <p className="projectStartDate">
              <span className="property">Start date: </span>
              {new Date(project.start_date).toDateString()}
            </p>
          ) : null}
          {project.end_date ? (
            <p className="projectEndDate">
              <span className="property">End date: </span>
              {new Date(project.end_date).toDateString()}
            </p>
          ) : null}
          <p className="projectTypeOfPayment">
            <span className="property">Type of payment: </span>
            {type_of_payment} 
          </p>
          <p className="projectStatus">
            <span className="property">Status:</span>
            {project.status ? "" : "not"} active
          </p>
        </div>

        <div className="btn-group-horizontal">
          <button
            type="button"
            className="btn btn-success btn-lg"
            onClick={() => getTotal(projectId)}
          >
            Total
          </button>

          <Link to={`${projectId}/update`}>
            <button type="button" className="updateProject btn btn-warning btn-lg">
              Update project
            </button>
          </Link>

          <button
            type="button"
            className="removeProject btn btn-danger btn-lg"
            onClick={() => removeProject(projectId)}
          >
            Delete project
          </button>
        </div>

        <div className="session">{sessionsType}</div>
      </div>
    );
  }
}

export default Project;
