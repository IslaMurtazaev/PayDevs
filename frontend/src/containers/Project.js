import { connect } from "react-redux";
import { projectActions } from "../actions/project";

import Project from "../components/Project";

const mapStateToProps = (state, ownProps) => ({
  project: state.project
    ? state.project
    : state.projects.find(project => project.id === +ownProps.match.params.id)
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  removeProject: id => {
    dispatch(projectActions.remove(id));
  },
  getTotal: id => {
    dispatch(projectActions.getTotal(id));
  },
  getProject: () => {
    dispatch(projectActions.get(+ownProps.match.params.id));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Project);
