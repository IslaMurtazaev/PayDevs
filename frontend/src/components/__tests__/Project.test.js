import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";
import toJson from "enzyme-to-json";
jest.mock("../../../index.js", () => require("history"));

import Project from "../../containers/Project";

describe("<Project />", () => {
  const initialState = {
    project: {
      id: 1,
      title: "PayDevs",
      description: "blablabla",
      start_date: "2018-08-24T00:00:00.000Z",
      end_date: "2018-09-30T10:00:00.000Z",
      type_of_payment: "M_P",
      status: true
    }
  };
  const mockStore = configureStore();

  let store = mockStore(initialState);
  let container = shallow(<Project store={store} match={ {params: {id: 1}} } />).dive();

  it("renders projectTitle based on props", () => {
    expect(container.find(".projectTitle").text()).toBe("PayDevs");
  });

  it("renders projectDescription based on props", () => {
    expect(container.find(".projectDescription").text()).toBe(
      "Description: blablabla"
    );
  });

  it("renders projectStartDate based on props", () => {
    expect(container.find(".projectStartDate").text()).toBe(
      "Start date: Fri Aug 24 2018"
    );
  });

  it("renders projectEndDate based on props", () => {
    expect(container.find(".projectEndDate").text()).toBe(
      "End date: Sun Sep 30 2018"
    );
  });

  it("renders projectTypeOfPayment based on props", () => {
    expect(container.find(".projectTypeOfPayment").text()).toBe(
      "Type of payment: Monthly"
    );
  });

  it("renders projectStatus based on props", () => {
    expect(container.find(".projectStatus").text()).toBe("Status: active");
  });

  it("renders three buttons", () => {
    expect(container.find("button")).toHaveLength(3);
  })

  // it("dispatches right action on .removeProject", () => {
  //   container.find(".removeProject").simulate("click");
  //   const actions = store.getActions();
  //   expect(actions).toEqual([ { type: 'REMOVE_PROJECT', projectId: 1 } ]);
  // })
  // it("renders buttons and connects then correctly", () => {
  //   const removeProjectSpy = jest.fn();
  //   container = shallow(<ProjectItem store={store} removeProject={removeProjectSpy} />).dive();
  //   container.find(".removeProject").simulate("click");
  //   expect(removeProjectSpy).toBeCalledWith(1);
  // })

  it("matches previous snap", () => {
    expect(toJson(container)).toMatchSnapshot();
  })
});
