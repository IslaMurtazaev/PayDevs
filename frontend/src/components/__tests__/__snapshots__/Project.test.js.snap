// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Project /> matches previous snap 1`] = `
<div
  className="container"
>
  <h1
    className="projectTitle"
  >
    PayDevs
  </h1>
  <div
    className="properties"
  >
    <p
      className="projectDescription"
    >
      <span
        className="property"
      >
        Description: 
      </span>
      blablabla
    </p>
    <p
      className="projectStartDate"
    >
      <span
        className="property"
      >
        Start date: 
      </span>
      Fri Aug 24 2018
    </p>
    <p
      className="projectEndDate"
    >
      <span
        className="property"
      >
        End date: 
      </span>
      Sun Sep 30 2018
    </p>
    <p
      className="projectTypeOfPayment"
    >
      <span
        className="property"
      >
        Type of payment: 
      </span>
      Monthly
    </p>
    <p
      className="projectStatus"
    >
      <span
        className="property"
      >
        Status:
      </span>
       active
    </p>
  </div>
  <div
    className="btn-group-horizontal"
  >
    <button
      className="btn btn-success btn-lg"
      onClick={[Function]}
      type="button"
    >
      Total
    </button>
    <Link
      replace={false}
      to="1/update"
    >
      <button
        className="updateProject btn btn-warning btn-lg"
        type="button"
      >
        Update project
      </button>
    </Link>
    <button
      className="removeProject btn btn-danger btn-lg"
      onClick={[Function]}
      type="button"
    >
      Delete project
    </button>
  </div>
  <div
    className="session"
  >
    <Connect(MonthlyRates)
      projectId={1}
    />
  </div>
</div>
`;
